src/library/scala/collection/immutable/Range.scala

```diff
@@ -165,6 +165,41 @@ sealed abstract class Range(
     }
   }
 
+  override final def indexOf[@sp(Int) B >: Int](elem: B, from: Int = 0): Int =
+    elem match {
+      case i: Int =>
+        val pos = posOf(i)
+        if (pos >= from) pos else -1
+      case _ => super.indexOf(elem, from)
+    }
+
+  override final def lastIndexOf[@sp(Int) B >: Int](elem: B, end: Int = -1): Int =
+    elem match {
+      case i: Int =>
+        val pos = posOf(i)
+        if (pos <= end) pos else -1
+      case _ => super.lastIndexOf(elem, end)
+    }
+
+  private[this] def posOf(i: Int): Int =
+    if (contains(i)) (i - start) / step else -1
+
+  override def sorted[B >: Int](implicit ord: Ordering[B]): IndexedSeq[Int] =
+    if (ord eq Ordering.Int) this else super.sorted(ord)
```

Comment:
Is it worth also check if `ord eq Ordering.Int.Reverse` to return `this.reverse` ?

Reply:
Maybe so. Anyone else have an opinion? 

The main reason I did this is that `Ordering.Int` is the vastly most common value that this method will be passed, and it's also the one for which the least work can be done. Passing `Int.Reverse` is probably far rarer.

Reply:
Maybe `Ordering.Int` could also `override val reverse: Ordering[Int] = ...` so that it doesn't need to allocate a new instance each time